### PR TITLE
Don't use `--add-modules` with GraalVM < 22.0.0.2

### LINKED ISSUE #28475: Module support not available in GraalVM / Mandrel 21.3
### Describe the bug

https://github.com/quarkusio/quarkus/pull/28409 adds support for modules. However, modules are not supported in GraalVM / Mandrel 21.3.

### Expected behavior

Quarkus native integration tests should pass when using 21.3.

### Actual behavior

Quarkus integartion tests fail when using 21.3. The error is:
```
--add-modules=jdk.net
```

See https://github.com/graalvm/mandrel/actions/runs/3216808337/jobs/5259259874#step:11:693

### How to Reproduce?

Run quarkus integration tests with `-Dquarkus.native.builder-image=quay.io/quarkus/ubi-quarkus-mandrel:21.3-java11`

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

11

### GraalVM version (if different from Java)

21.3

### Quarkus version or git rev

06c4344e6b203a798eb73f01e6e1d70fe01815b3

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by zakkak**
Support for `--add-modules` was added in 22.0.0.2
https://github.com/oracle/graal/commit/f7b98697f1506ddb4cd317aa2b1b57748d03d986

Fixes #28475

### CODE REVIEW

**Review by Sanne — APPROVED** [review]
thanks!

**Review by gsmet — CHANGES_REQUESTED** [review]
I wonder if we should change the minimal version instead?

I think that would make sense given some things won't work very well without this.

**Comment by Sanne** [comment]
I'm in favour to bump the minimal version, but I'm not sure about "some things won't work very well without this" - as older GraalVM versions wouldn't run `native-image` on the module path, so it's possible that the (single) case in which I figured we need to use `--add-modules` isn't actually necessary on the older version (and it would work).

That might need testing; but indeed if we were to bump minimal GraalVM version we're more likely to spare some trouble.

**Comment by Sanne** [comment]
And FYI this is the only conumer for this builditem:
https://github.com/quarkusio/quarkus/blob/141753f9d4307b8ba0a0ab14c9f0e74e03af90be/extensions/jdbc/jdbc-h2/deployment/src/main/java/io/quarkus/jdbc/h2/deployment/JDBCH2Processor.java#L60-L64

**Comment by Sanne** [comment]
I'll let @zakkak decide. Personally I'd love to drop the old GraalVM versions but I'm not sure of the implications - either way I believe we could merge this one first, and then drop them... up to him :)

**Comment by zakkak** [comment]
+1 on dropping support for 21.3 as there will be no further 21.3.x releases moving forward (i.e. it's reaching its EOL on the 25th of October).

If Quarkus 2.14 is not going to be released before the 25th of October then I would suggest bumping the minimum to 22.2 and the current to 22.3 (once released).

If bumping the minimum to 22.2 (released on the 26th of July) seems too drastic we can bump it to 22.1 instead.

Regarding Quarkus 2.13 the plan is to support 22.3 there as well, but to be on the safe side I would prefer keeping it compatible with 21.3 as well.

cc @jerboaa @Karm

**Comment by jerboaa** [comment]
> Regarding Quarkus 2.13 the plan is to support 22.3 there as well, but to be on the safe side I would prefer keeping it compatible with 21.3 as well.
> 
> cc @jerboaa @Karm

I don't think much testing has been carried out with Quarkus 2.13 and graal/mandrel 21.3, so I'm not convinced we need to keep it (or rather this fix is worth doing). There is a cost to actually ensure it (2.13 + 21.3) in addition to the other, more recent graal versions (22.x). We should have by-in from the quarkus devs on that. It looks like if we really have to, making it compatible to 21.3 would be fairly straight forward (at least on the surface).

IMO, minimal 22.2 and default 22.3 once it releases later this month would be my preference.

**Comment by zakkak** [comment]
> I don't think much testing has been carried out with Quarkus 2.13 and graal/mandrel 21.3, so I'm not convinced we need to keep it (or rather this fix is worth doing).

FWIW this combination was actually being tested nightly till the release of Quarkus 2.13, so I would say that it has been tested quiet well up to a point. Nevertheless, I am OK with not fixing https://github.com/quarkusio/quarkus/issues/28530 (note that this issue that we are commenting in does not affect Quarkus 2.13)

**Comment by gsmet** [comment]
I'm +1 for making 22.2 the minimum version in 2.14.

We used to be far more aggressive about that and I think it's fair to push people to the latest and greatest of GraalVM given how fast it evolves.

**Comment by gsmet** [comment]
Making it draft as if all goes well, we won't merge this one. Keeping it around for potential emergency plan though.

**Pull request closed without merging**
